components/ui/native-chain-progress.tsx

```
@@ -0,0 +1,193 @@
+                onStatusChange?.("success");
+                return { ...final, steps: finalSteps };
+              });
+            }, 3000);
+          }, 5000);
+          break;
+
+        case "error":
+          // Find the current processing step and mark it as error
+          const processingIndex = steps.findIndex(
+            (step) => step.status === "processing",
+          );
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)